addons/calendar/calendar.py

```diff
@@ -455,62 +446,44 @@ def get_next_mail(self, cr, uid, context=None):
+                last_found = self.do_check_alarm_for_one_date(cr, uid, in_date_format, curEvent, max_delta, 0, missing=True, context=context)
                 for alert in last_found:
-                    self.do_mail_reminder(cr, uid, alert, context=context)
-        icp.set_param(cr, SUPERUSER_ID, 'calendar.last_notif_mail', now)
-
-    def get_next_notif(self, cr, uid, context=None):
-        ajax_check_every_seconds = 300
-        partner = self.pool['res.users'].read(cr, SUPERUSER_ID, uid, ['partner_id', 'calendar_last_notif_ack'], context=context)
-        all_notif = []
-
-        if not partner:
-            return []
-
-        all_events = self.get_next_potential_limit_alarm(cr, uid, ajax_check_every_seconds, partner_id=partner['partner_id'][0], mail=False, context=context)
-
-        for event in all_events:  # .values()
-            max_delta = all_events[event]['max_duration']
-            curEvent = self.pool.get('calendar.event').browse(cr, uid, event, context=context)
-            if curEvent.recurrency:
-                bFound = False
-                LastFound = False
-                for one_date in self.pool.get("calendar.event").get_recurrent_date_by_event(cr, uid, curEvent, context=context):
-                    in_date_format = one_date.replace(tzinfo=None)
-                    LastFound = self.do_check_alarm_for_one_date(cr, uid, in_date_format, curEvent, max_delta, ajax_check_every_seconds, after=partner['calendar_last_notif_ack'], mail=False, context=context)
-                    if LastFound:
-                        for alert in LastFound:
-                            all_notif.append(self.do_notif_reminder(cr, uid, alert, context=context))
-                        if not bFound:  # if it's the first alarm for this recurrent event
-                            bFound = True
-                    if bFound and not LastFound:  # if the precedent event had alarm but not this one, we can stop the search fot this event
-                        break
-            else:
-                in_date_format = datetime.strptime(curEvent.start, DEFAULT_SERVER_DATETIME_FORMAT)
-                LastFound = self.do_check_alarm_for_one_date(cr, uid, in_date_format, curEvent, max_delta, ajax_check_every_seconds, after=partner['calendar_last_notif_ack'], mail=False, context=context)
-                if LastFound:
-                    for alert in LastFound:
+                    if alert['alarm_type'] == 'email':
+                        self.do_mail_reminder(cr, uid, alert, context=context)
+                    if alert['alarm_type'] == 'notification':
                         all_notif.append(self.do_notif_reminder(cr, uid, alert, context=context))
-        return all_notif
+        if all_notif:
+            self.pool['bus.bus'].sendmany(cr, uid, all_notif)
+        else:
+            now = openerp.fields.Datetime.to_string(datetime.now())
+            icp.set_param(cr, SUPERUSER_ID, 'calendar.last_notif_mail', now)
```

Comment:
It means if there are some notifications then do not set last_notif_mail param, why not always set last_notif_mail param ?